Reviewer/src/main/java/com/softserveinc/reviewer/health/TimeLimitedHealthCheck.java

```diff
@@ -0,0 +1,34 @@
+package com.softserveinc.reviewer.health;
+
+import java.util.Arrays;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.health.HealthCheck;
+
+public class TimeLimitedHealthCheck extends HealthCheck {
+    private final long timeout;
+    private final TimeUnit timeUnit;
+    private final Callable<Boolean>[] asyncHealthChecks;
+    private final ExecutorService executorService = Executors.newFixedThreadPool(4);
+
+    public TimeLimitedHealthCheck(long timeout, TimeUnit timeUnit, Callable<Boolean>... asyncHealthChecks) {
+        this.timeout = timeout;
+        this.timeUnit = timeUnit;
+        this.asyncHealthChecks = asyncHealthChecks;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        for (Future<Boolean> future : executorService.invokeAll(Arrays.asList(asyncHealthChecks), timeout, timeUnit)) {
+            if (future.isCancelled() || !future.get()) {
+                return Result.unhealthy("At least one of the Services are unhealthy");
```

Comment:
Please correct typo in the message. If you talk about one service, it should be `is`. Or you can even go without message at all. It doesn't provide any valuable information anyway.